packages/dds/task-manager/src/taskManager.ts

```diff
@@ -312,6 +312,34 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         return lockAcquireP;
     }
 
+    public subscribeToTask(taskId: string) {
+        const disconnectHandler = () => {
+            // Wait to be connected again and then re-submit volunteer op
+            this.connectionWatcher.once("connect", () => {
+                this.submitVolunteerOp(taskId);
+            });
+        };
+
+        const checkIfAbandoned = (eventTaskId: string) => {
+            if (eventTaskId !== taskId) {
+                return;
+            }
+
+            this.abandonWatcher.off("abandon", checkIfAbandoned);
+            this.connectionWatcher.off("disconnect", disconnectHandler);
+        };
+
+        this.abandonWatcher.on("abandon", checkIfAbandoned);
+        this.connectionWatcher.on("disconnect", disconnectHandler);
+
+        if (!this.connected) {
+            disconnectHandler();
```

Comment:
One thing I'm thinking about -- there's currently no way to detect that you are subscribed for a given task -- this is probably most noticeable in disconnected state, where you won't even be queued().  I'm wondering if we should have API surface that reveals subscribed status (not necessary for this PR probably)

Reply:
I think its fair to add it into this PR since it goes hand in hand with subscribeToTask().

Reply:
@ChumpChief, thoughts on implementation for this API? I was actually thinking it may be good to introduce a new "subscribe" op type for this. If that's the case it may be better to defer this to a separate PR to limit the size of this one. Let me know what you think though.

Reply:
I think sticking with just "volunteer" op is good - suppose client A sees a "subscribe" op from client B, and then they see client B disconnect.  Even if they see B reconnect, B will have a different clientId so A would have to do some work to reassociate B with the new clientId.  And B might never reconnect, so A wouldn't be able to conclude whether it should keep tracking B as subscribed (i.e. it could never be cleaned up from the summary).

With this feedback I was mostly just thinking about how a client detects itself as subscribed, e.g. so they can do stuff like

```ts
if (this.isSubscribed(taskId)) {
    this.abandon(taskId);
}

if (!this.isSubscribed(taskId)) {
    this.subscribeToTask(taskId);
}
```

It might also help simplify some state management internally as well.  E.g. it looks like currently, if I am disconnected and call `this.subscribeToTask(taskId)` three times, then on reconnect I will submit three volunteer ops.